### PR TITLE
#150507660 Update Notification Status

### PR DESCRIPTION
**Pull request by Mnickii**
#### What does this PR do?
-Update a notification status from unread to read.
#### Description of Task to be completed?
-Add update notifications status controller
-Add update notification status test

#### How should this be manually tested?
At the root of the project directory run the following command:
        
    $ pytest -v 

#### What are the relevant pivotal tracker stories?
#150507660

### CODE REVIEW

File: api/views/contributions.py [hunk 1]
```diff
@@ -409,3 +409,56 @@ def get(self, notif_id=None):
...
+            response.status_code= 200
+            return response  
+        else:
+            response = jsonify(dict(
+                message="Notification does not exist",
+                status=404
+            ))
+            response.status_code= 404
+            return response
+
```
Comment (Andela-ddiei) [inline api/views/contributions.py]: Add a new line to the end of the file.

File: test/api/views/test_contributions.py [hunk 2]
```diff
@@ -1396,3 +1396,24 @@ def test_non_admin_get_single_notification(self):
             response_data["status_code"], 401
         )
         self.assertEqual(response_data["message"], expected_message)
+
+    def test_update_notification_status(self):
+        fetched_notif = Notification.query.all()[0]
+        notif_id = fetched_notif.id
+        print('id', fetched_notif.status)
```
Comment (Andela-ddiei) [inline test/api/views/test_contributions.py]: Remove print statement

File: test/api/views/test_contributions.py [hunk 3]
```diff
@@ -1396,3 +1396,24 @@ def test_non_admin_get_single_notification(self):
...
+        )
+        new_notif_status = Notification.query.get(notif_id).status
+        expected_message = "Notification updated!"
+        generated_response = json.loads(response.data)
+
+        self.assertEqual(response.status_code, 200),
+        self.assertEqual(expected_message,
+                         generated_response['message'])
+        self.assertNotEqual(old_notif_status, new_notif_status)
+
```
Comment (Andela-ddiei) [inline test/api/views/test_contributions.py]: Add new line

Comment (Mnickii) [reply test/api/views/test_contributions.py]: Done

File: test/api/views/test_contributions.py (hunk 2 above)
Comment (Mnickii) [reply test/api/views/test_contributions.py]: Done

File: api/views/contributions.py (hunk 1 above)
Comment (Mnickii) [reply api/views/contributions.py]: DOne

**Comment by LarryWachira** [comment]
Rebase this @Mnickii so we can review.

**Comment by Mnickii** [comment]
Done @LarryWachira

File: api/views/contributions.py [hunk 4]
```diff
@@ -392,3 +392,55 @@ def get(self, notif_id=None):
...
+            ))
+            response.status_code = 400
+            return response
+
+        data = request.get_json()
+
+        notification = Notification.query.get(notif_id)
+
+        if notification:
+            if "status" in data:
```
Comment (LarryWachira) [inline api/views/contributions.py]: I'm curious, if status is not in data, what happens?

File: api/views/contributions.py [hunk 5]
```diff
@@ -402,3 +402,54 @@ def get(self, notif_id=None):
...
                             previousUrl=previous_url,
                             currentPage=_notifications.page))
+
+    @token_required
+    @admin_required
+    def patch(self, notif_id):
+        
+        if not request.json:
+            response = jsonify(dict(
+                status=400,
```
Comment (LarryWachira) [inline api/views/contributions.py]: Consider refactoring the responses to match JSEND specification.

**Comment by LarryWachira** [comment]
Also, rebase and squash.

File: api/views/contributions.py (hunk 5 above)
Comment (Mnickii) [reply api/views/contributions.py]: Done!

**Comment by andela-oowonikoko** [comment]
@Mnickii Please rebase your branch and ensure your commits are squashed :thank_you:

**Comment by Mnickii** [comment]
This is No longer relevant due to the refactored Notification system.

**Pull request closed without merging**